src/StreamrClient.js

```diff
@@ -276,8 +272,23 @@ export default class StreamrClient extends EventEmitter {
             const streamMessage = await this.msgCreationUtil.createStreamMessage(stream, data, timestamp, partitionKey)
             return this._requestPublish(streamMessage, sessionToken)
         } else if (this.options.autoConnect) {
-            this.publishQueue.push([streamId, data, timestamp, partitionKey])
-            return this.connect().catch(() => {}) // ignore
+            if (this.publishQueue.length >= this.options.maxPublishQueueSize) {
+                throw new FailedToPublishError(
+                    streamId,
+                    data,
+                    `publishQueue exceeded maxPublishQueueSize=${this.options.maxPublishQueueSize}`,
+                )
+            }
```

Comment:
Optional, but I feel it would be polite to provide some API that could be used to check before the user calls publish? Otherwise it seems like the only way to work around this is to wait for this exception, hope you detected this specific exception correctly and respond from there i.e. exception based flow control.

This is to catch edge cases and ideally users would raise their max queue size appropriately but it's also possible they aren't in control of how many messages are being published. I guess in those cases they could set the queue size to `Infinity`.